EDOT PHP documentation (#144)
elastic/opentelemetry: docs/_edot-sdks/php/supported-technologies.md

parent: EDOT PHP
---

> # The Elastic Distribution of OpenTelemetry PHP supports the following technologies

Reviewer: ```suggestion
# EDOT PHP supports the following technologies
```

otherwise the title is too long for a <h1>

> This allows you to capture **detailed traces** from libraries such as **cURL**, **PDO**, and **MySQLi**, even in PHP 8.1 environments.

## Supported PHP SAPI's
- php-cli
- php-fpm
- php-cgi/fcgi
- mod_php (prefork)

EDOT PHP supports all popular variations of using PHP in combination with a web server, such as Apache + mod_php, Apache + php_fpm or cgi, NGINX + php_fpm or cgi, and others.

## Supported Operating Systems
- **Linux**
  - Architectures: **x86_64** and **ARM64**
  - **glibc-based systems**: Packages available as **DEB** and **RPM**
  - **musl libc-based systems (Alpine Linux)**: Packages available as **APK**

## Instrumented Frameworks
- Laravel (v6.x/v7.x/v8.x/v9.x/v10.x/v11.x)
- Slim (v4.x)

## Instrumented Libraries
- Curl (v8.1 - v8.4)
- HTTP Async (php-http/httplug v2.x)
- MySQLi (v8.1 - v8.4)
- PDO (v8.1 - v8.4)

## Extends the upstream OpenTelemetry PHP with additional features and improvements
- **Truly zero-config auto-instrumentation**
  Unlike the upstream OpenTelemetry PHP agent, **EDOT PHP works fully automatically** — no need to modify your application code, add Composer packages, or adjust deployment scripts.

>   Once the system package is installed, EDOT PHP automatically detects your application and injects the instrumentation code at runtime, with **no manual integration required**.

Reviewer: ```suggestion
  Once the system package is installed, EDOT PHP automatically detects your application and injects the instrumentation code at runtime, **without a need for manual integration**.
```
